org_fedora_hello_world/dbus_service/hello_world.py

```diff
@@ -0,0 +1,110 @@
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.core.configuration.anaconda import conf
+from pyanaconda.core.dbus import DBus
+from pyanaconda.core.signal import Signal
+from pyanaconda.modules.common.base import KickstartService
+from pyanaconda.modules.common.containers import TaskContainer
+
+from org_fedora_hello_world.constants import HELLOWORLD
+from org_fedora_hello_world.dbus_service.hello_world_interface import HelloWorldInterface
+from org_fedora_hello_world.dbus_service.installation import \
+    HelloWorldCalculationTask, HelloWorldConfigurationTask, HelloWorldInstallationTask
+from org_fedora_hello_world.dbus_service.kickstart import HelloWorldKickstartSpecification
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class HelloWorld(KickstartService):
+    """The HelloWorld service.
+
+    This class parses and stores data for the Hello world addon.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        self._seen = False
+        self._reverse = False
+        self._lines = []
+
+        self.reverse_changed = Signal()
+        self.lines_changed = Signal()
+
+    def publish(self):
+        """Publish the module."""
+        TaskContainer.set_namespace(HELLOWORLD.namespace)
+        DBus.publish_object(HELLOWORLD.object_path, HelloWorldInterface(self))
+        DBus.register_service(HELLOWORLD.service_name)
+
+    @property
+    def kickstart_specification(self):
+        """Return the kickstart specification."""
+        return HelloWorldKickstartSpecification
+
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        log.debug("Processing kickstart data...")
+        self._seen = data.addons.org_fedora_hello_world.seen
+        self._reverse = data.addons.org_fedora_hello_world.reverse
+        self._lines = data.addons.org_fedora_hello_world.lines
+
+    def setup_kickstart(self, data):
+        """Set the given kickstart data."""
+        log.debug("Generating kickstart data...")
+        data.addons.org_fedora_hello_world.seen = self._seen
+        data.addons.org_fedora_hello_world.reverse = self._reverse
+        data.addons.org_fedora_hello_world.lines = self._lines
+
+    @property
+    def reverse(self):
+        return self._reverse
+
+    def set_reverse(self, reverse):
+        self._reverse = reverse
+        self.reverse_changed.emit()
+        log.debug("Reverse is set to %r.", reverse)
+
+    @property
+    def lines(self):
+        return self._lines
+
+    def set_lines(self, lines):
+        self._lines = lines
+        self.lines_changed.emit()
+        log.debug("Lines is set to %r.", lines)
+
+    def configure_with_tasks(self):
+        """Return configuration tasks."""
```

Comment:
These tasks will be run by default at the beginning of the installation process.